awssqs/src/main/java/zipkin/reporter/awssqs/AwsBufferedSqsSender.java

```diff
@@ -0,0 +1,235 @@
+    return builder().queueUrl(url).build();
+  }
+
+  public static Builder builder() {
+    return new AutoValue_AwsBufferedSqsSender.Builder()
+        .encoding(Encoding.THRIFT)
+        .compressionEnabled(true)
+        .awsSecretAccessKey(null)
+        .awsAccessKeyId(null)
+        .awsRoleName(null)
+        .awsServiceId(null)
+        .maxRequests(64)
+        .maxQueuedRequests(1000)
+        .messageMaxBytes(256 * 1024 - 30); // 256KB SQS limit per message. 30 bytes for attr keys.
+  }
+
+  @AutoValue.Builder
+  public static abstract class Builder {
+
+    public abstract Builder queueUrl(String queueUrl);
+    public abstract Builder awsAccessKeyId(String awsAccessKeyId);
+    public abstract Builder awsSecretAccessKey(String awsSecretAccessKey);
+    public abstract Builder awsRoleName(String awsRoleName);
+    public abstract Builder awsServiceId(String awsServiceId);
+    public abstract Builder compressionEnabled(boolean compressionEnabled);
+    public abstract Builder maxRequests(int maxRequests);
+    public abstract Builder maxQueuedRequests(int maxQueuedRequests);
+
+
+    public abstract Builder encoding(Encoding encoding);
+
+    /**
+     * Maximum size of a message. SQS max message size is 256KB including attributes.
+     * Default 256 less 30 bytes for message keys.
+     *
+     * @param messageMaxBytes
+     * @return
+     */
+    public abstract Builder messageMaxBytes(int messageMaxBytes);
+
+    public final AwsBufferedSqsSender build() {
+      return autoBuild();
+    }
+
+    abstract AwsBufferedSqsSender autoBuild();
+
+    Builder() {}
+  }
+
+  public Builder toBuilder() {
+    return new AutoValue_AwsBufferedSqsSender.Builder(this);
+  }
+
+  abstract String queueUrl();
+  @Nullable abstract String awsAccessKeyId();
+  @Nullable abstract String awsSecretAccessKey();
+  @Nullable abstract String awsRoleName();
+  @Nullable abstract String awsServiceId();
+  abstract boolean compressionEnabled();
+  abstract int maxRequests();
+  abstract int maxQueuedRequests();
+
+  /** close is typically called from a different thread */
+  private AtomicBoolean closeCalled = new AtomicBoolean(false);
+
+
+  @Override public CheckResult check() {
+    try {
+      List<String> queueUrls = get().listQueues().getQueueUrls();
+      if (queueUrls.contains(queueUrl())) {
+        return CheckResult.OK;
+      } else {
+        return CheckResult.failed(new IllegalStateException(String.format("queue url not found %s", queueUrl())));
+      }
+    } catch (Exception e) {
+      return CheckResult.failed(e);
+    }
+  }
+
+  @Override protected AmazonSQSBufferedAsyncClient compute() {
+    // bound the executor so that we get consistent performance
+    ThreadPoolExecutor dispatchExecutor =
+        new ThreadPoolExecutor(0, maxRequests(), 60, TimeUnit.SECONDS,
+            new ArrayBlockingQueue<>(maxQueuedRequests()),
+            threadFactory("AwsBufferedSqsSender Dispatcher", false));
```

Comment:
The `ThreadPoolExecutor`, if we provide a default one beyond the default one included in the aws-sdk then it should probably be configurable